src/components/Reports/GDPRPermissions/PermissionExpandableRow.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from 'react';
+import * as S from './SubscriberGdpr.styles';
+import { FormattedMessage, useIntl, FormattedDate } from 'react-intl';
+import PropTypes from 'prop-types';
+import { Loading } from '../../Loading/Loading';
+import { InjectAppServices } from '../../../services/pure-di';
+
+const PermissionValue = ({ value }) => {
+  const permissionValue = value.toLowerCase();
+  let iconColor =
+    permissionValue === 'none' ? 'grey' : permissionValue === 'true' ? 'green' : 'red';
+
+  return (
+    <div className="dp-icon-wrapper">
+      <span className={`ms-icon icon-lock dp-lock-${iconColor}`} />
```

Comment:
I'm not so sure about this approach, I think the previous idea without variables makes it more clear which state and what ui class we are using. @lucianaDilizio @andresmoschini  what do you think?

Reply:
I am not sure to understand your approach @cbernat, let me confirm with an example. Are you talking about something like this?

```suggestion
                        {field.value.toLowerCase() === 'none' ? (
                          <div className="dp-icon-wrapper">
                            <span className="ms-icon icon-lock dp-lock-grey"></span>
                            <FormattedMessage id="subscriber_gdpr.value_none" />
                          </div>
                        ) : field.value.toLowerCase() === 'true' ? (
                          <div className="dp-icon-wrapper">
                            <span className="ms-icon icon-lock dp-lock-green"></span>
                            <FormattedMessage id="subscriber_gdpr.value_true" />
                          </div>
                        ) : (
                          <div className="dp-icon-wrapper">
                            <span className="ms-icon icon-lock dp-lock-red"></span>
                            <FormattedMessage id="subscriber_gdpr.value_false" />
                          </div>
                        )}
```

Reply:
I like @bravoy93's implementation, it reduced the duplicated code. Please @cbernat let us know if you have a strong opinion on this.